fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncResultsNotReadySpec.groovy

```diff
@@ -0,0 +1,71 @@
+package com.yahoo.bard.webservice.async
+
+import com.yahoo.bard.webservice.async.jobs.jobrows.DefaultJobField
+import com.yahoo.bard.webservice.async.jobs.jobrows.DefaultJobStatus
+
+import spock.lang.Timeout
+
+import javax.ws.rs.core.Response
+
+/**
+ * Verifies that if the result of an asynchronous query is not ready yet, then the user get back the exact same payload
+ * from the jobs endpoint they received from the data endpoint.
+ */
+@Timeout(30)
+class AsyncResultsNotReadySpec extends AsyncFunctionalSpec {
+
+    static final String QUERY =
+            "http://localhost:9998/data/shapes/day/color?dateTime=2016-08-30%2F2016-08-31&metrics=height&asyncAfter=0"
+
+    @Override
+    Map<String, Closure<String>> getResultsToTargetFunctions() {
+        return [
+                data: {"data/shapes/day/color"},
+                jobs: {AsyncTestUtils.buildTicketLookup(it.data.readEntity(String))},
+                results: {payloads ->
+                    String dataPayload = payloads.data.readEntity(String)
+                    "jobs/${AsyncTestUtils.getJobFieldValue(dataPayload, DefaultJobField.JOB_TICKET.name)}/results"
+                }
+        ]
+    }
+
+    @Override
+    Map<String, Closure<Void>> getResultAssertions() {
+        return [
+                data: {Response response ->
+                    assert response.status == 202
+                    AsyncTestUtils.validateJobPayload(response.readEntity(String), QUERY, DefaultJobStatus.PENDING.name)
+                },
+                jobs: {Response response ->
+                    assert response.status == 200
+                    AsyncTestUtils.validateJobPayload(response.readEntity(String), QUERY, DefaultJobStatus.PENDING.name)
+                },
+                results: {Response response ->
+                    assert response.status == 200
+                    AsyncTestUtils.validateJobPayload(response.readEntity(String), QUERY, DefaultJobStatus.PENDING.name)
```

Comment:
Yeah, originally I was going to do that, but it ended up being easier just to use the same validation method with the same arguments. Since the validation method is side-effect free, I _am_ verifying that they're the same. ;)